jobs/scrape_archives.py

```diff
@@ -295,24 +295,34 @@ def is_release_metadata(product, version, filename):
     listings = await asyncio.gather(*futures)
 
     channel = "nightly"
+    batch_size = 10
 
     for day_url, (_, files) in zip(days_urls, listings):
-        for file_ in files:
-            filename = file_["name"]
-
-            try:
-                version, locale, platform = parse_nightly_filename(filename)
-            except ValueError:
-                continue
-
-            size = file_["size"]
-            date = file_["last_modified"]
-            url = day_url + filename
-
-            metadata = await fetch_nightly_metadata(session, url)
-            record = archive(product, version, platform, locale, channel, url, size, date, metadata)
-            logger.debug("Nightly found {}".format(url))
-            await queue.put(record)
+        # Fetch metadata and put into queue in batch.
+        nb_batches = (len(files) // batch_size) + 1
+        for i in range(nb_batches):
+            files_subset = files[(i * batch_size):((i + 1) * batch_size)]
```

Comment:
This is kind of clunky. Too bad there's no `itertools.chunks` in the stdlib. Maybe use https://boltons.readthedocs.io/en/latest/iterutils.html#boltons.iterutils.chunked ?